pkg/process/runner/runner.go

```diff
@@ -393,6 +398,53 @@ func (l *CheckRunner) basicRunner(c checks.Check) func() {
 	}
 }
 
+// getConnectionsCheckInterval validates 'process_config.connections_capacity_check_interval' (the "capacity interval").
+// It must be between 10s-30s (inclusive) and be a factor of defaultInterval (the main connections check interval).
+// If both are true, the capacity interval is returned.
+// If explicitly configured by the user but fails validation, a specific warning is logged and defaultInterval is returned.
+// Otherwise (e.g. not configured, or derived value is invalid), defaultInterval is returned.
+func (l *CheckRunner) getConnectionsCheckInterval(checkName string, defaultInterval time.Duration) time.Duration {
```

Comment:
do we have UTs for this?

Reply:
I simplified this greatly and moved to `interval.go`. added UTs as well